spec/requests/internal/configs_spec.rb

```diff
@@ -15,137 +17,166 @@
       end
     end
 
-    describe "Google Analytics Reporting API v4" do
-      it "updates ga_view_id" do
-        post "/internal/config", params: { site_config: { ga_view_id: "abc" } }
-        expect(SiteConfig.ga_view_id).to eq("abc")
-      end
-
-      it "updates ga_fetch_rate" do
-        post "/internal/config", params: { site_config: { ga_fetch_rate: 3 } }
-        expect(SiteConfig.ga_fetch_rate).to eq(3)
-      end
-    end
-
-    describe "Mailchimp lists IDs" do
-      it "updates mailchimp_newsletter_id" do
-        post "/internal/config", params: { site_config: { mailchimp_newsletter_id: "abc" } }
-        expect(SiteConfig.mailchimp_newsletter_id).to eq("abc")
-      end
-
-      it "updates mailchimp_sustaining_members_id" do
-        post "/internal/config", params: { site_config: { mailchimp_sustaining_members_id: "abc" } }
-        expect(SiteConfig.mailchimp_sustaining_members_id).to eq("abc")
-      end
-
-      it "updates mailchimp_tag_moderators_id" do
-        post "/internal/config", params: { site_config: { mailchimp_tag_moderators_id: "abc" } }
-        expect(SiteConfig.mailchimp_tag_moderators_id).to eq("abc")
-      end
-
-      it "updates mailchimp_community_moderators_id" do
-        post "/internal/config", params: { site_config: { mailchimp_community_moderators_id: "abc" } }
-        expect(SiteConfig.mailchimp_community_moderators_id).to eq("abc")
-      end
-    end
-
-    describe "Email digest frequency" do
-      it "updates periodic_email_digest_max" do
-        post "/internal/config", params: { site_config: { periodic_email_digest_max: 1 } }
-        expect(SiteConfig.periodic_email_digest_max).to eq(1)
-      end
-
-      it "updates periodic_email_digest_min" do
-        post "/internal/config", params: { site_config: { periodic_email_digest_min: 3 } }
-        expect(SiteConfig.periodic_email_digest_min).to eq(3)
-      end
-    end
-
-    describe "Tags" do
-      it "removes space suggested_tags" do
-        post "/internal/config", params: { site_config: { suggested_tags: "hey, haha,hoho, bobo fofo" } }
-        expect(SiteConfig.suggested_tags).to eq(%w[hey haha hoho bobofofo])
-      end
-
-      it "downcases suggested_tags" do
-        post "/internal/config", params: { site_config: { suggested_tags: "hey, haha,hoHo, Bobo Fofo" } }
-        expect(SiteConfig.suggested_tags).to eq(%w[hey haha hoho bobofofo])
+    context "when admin has full permissions" do
+      before do
+        sign_in(admin_plus_config)
+      end
+      describe "staff" do
+        it "does not allow the staff_user_id to be updated" do
+          expect(SiteConfig.staff_user_id).to eq(1)
+          post "/internal/config", params: { site_config: { staff_user_id: 2 }, confirmation: confirmation_message }
+          expect(SiteConfig.staff_user_id).to eq(1)
+        end
```

Comment:
do we want to add this back in to the config form? or will that be a separate PR?

Reply:
I feel like we can make this a separate thing if we want to. Might be nice to sit on this before making the final choice.

Reply:
that sounds good to me 👍

Reply:
@benhalpern I'm looking at this code again, and I'm wondering if we meant for the `staff_user_id` to _not_ be updated, even if you pass it a `confirmation_message`? If someone passes a confirmation message, we should allow for this to be updated, right?

Just pairing with @juliannatetreault on adding `mascot_user_id` into the SiteConfig right now and noticed that `staff_user_id` wasn't in the params, so wanted to double-check the intent here!